SharpMap/Layers/Layer.cs

```diff
@@ -280,8 +285,14 @@ public virtual int TargetSRID
             get { return _targetSrid.HasValue ? _targetSrid.Value : SRID; }
             set
             {
-                _targetSrid = value;
-                _targetFactory = GeoAPI.GeometryServiceProvider.Instance.CreateGeometryFactory(value);

```

Comment:
Better?:
```C#
if (value == SRID) {
    _targetSrid = null;
    _targetFactory = null;
} else if (_targetSrid != value) {
    _targetSrid = value;
    _targetFactory = GeoAPI.GeometryServiceProvider.Instance.CreateGeometryFactory(value);
}
if (!_shouldNotResetCt)
    _coordinateTransform = _reverseCoordinateTransform = null;
```

Reply:
Yes - incorporated into latest commit. Have kept **TargetSRID** = 0 (zero) instead of null as zero has been used consistently up to now (eg [NeedsTransformation](https://github.com/SharpMap/SharpMap/blob/Branches/1.0/SharpMap/Layers/Layer.cs#L234)).